cmd/peek-submission/main.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+
+	"github.com/alecthomas/kong"
+
+	"github.com/edulinq/autograder/internal/api/core"
+	"github.com/edulinq/autograder/internal/api/courses/assignments/submissions"
+	"github.com/edulinq/autograder/internal/common"
+	"github.com/edulinq/autograder/internal/config"
+	"github.com/edulinq/autograder/internal/log"
+	"github.com/edulinq/autograder/internal/util"
+)
+
+var args struct {
+	config.ConfigArgs
+	TargetEmail      string `help:"Email of the user to fetch." arg:""`
+	CourseID         string `help:"ID of the course." arg:""`
+	AssignmentID     string `help:"ID of the assignment." arg:""`
+	TargetSubmission string `help:"ID of the submission." arg:"" optional:""`
+	Verbose          bool   `help:"Print the entire response." short:"v"`
+}
+
+func main() {
+	kong.Parse(&args,
+		kong.Description("Fetch a submission for a specific assignment and user."),
+	)
+
+	err := config.HandleConfigArgs(args.ConfigArgs)
+	if err != nil {
+		log.Fatal("Failed to load config options.", err)
+	}
+
+	socketPath, err := common.GetUnixSocketPath()
+	if err != nil {
+		log.Fatal("Failed to get the unix socket path.", err)
+	}
+
+	conn, err := net.Dial("unix", socketPath)
+	if err != nil {
+		log.Fatal("Failed to dial the unix socket.", err)
+	}
+	defer conn.Close()
+
+	request := submissions.FetchUserPeekRequest{
+		APIRequestAssignmentContext: core.APIRequestAssignmentContext{
+			APIRequestCourseUserContext: core.APIRequestCourseUserContext{
+				CourseID: args.CourseID,
+			},
+			AssignmentID: args.AssignmentID,
+		},
+		TargetUser: core.TargetCourseUserSelfOrGrader{
+			TargetCourseUser: core.TargetCourseUser{
+				Email: args.TargetEmail,
+			},
+		},
+		TargetSubmission: args.TargetSubmission,
+	}
+
+	requestMap := map[string]interface{}{
+		"endpoint": core.NewEndpoint(`courses/assignments/submissions/fetch/user/peek`),
+		"request":  request,
+	}
+
+	jsonRequest := util.MustToJSONIndent(requestMap)
```

Comment:
This is fine for now.
Note that it will need to transitioned to a non-Must function when generalized.